sample-tracking-status-overview-domain/src/main/groovy/life/qbic/business/samples/count/CountSamples.groovy

```diff
@@ -48,12 +49,20 @@ class CountSamples implements CountSamplesInput{
     }
   }
 
-  private int countReceivedSamplesFromStatus(List<Status> sampleStatuses) {
-    int receivedIndex = statusesInOrder.indexOf(Status.SAMPLE_RECEIVED)
-    // statuses that are not considered in the ordered list return -1, meaning the sample is not counted
-    return sampleStatuses.findAll {statusesInOrder.indexOf(it) >= receivedIndex}.size()
+  @Override
+  void countQcFailedSamples(String projectCode) {
+    try {
+      sampleStatuses = dataSource.fetchSampleStatusesForProject(projectCode)
+      int receivedAmount = countSamplesFromStatus(Status.SAMPLE_QC_FAIL)
+      output.countedFailedQcSamples(projectCode, sampleStatuses.size(), receivedAmount)
+    }catch (Exception e) {
+      output.failedExecution("Could not count failed qc samples.")
+    }
   }
 
-
-
-}
+  private int countSamplesFromStatus(Status status) {
+    int receivedIndex = statusesInOrder.indexOf(status)
+    // statuses that are not considered in the ordered list return -1, meaning the sample is not counted
+    return sampleStatuses.findAll { statusesInOrder.indexOf(it) >= receivedIndex }.size()
+  }
```

Comment:
Are FAILED_QC samples always stuck at this point? This method counts all samples that are failedqc or further along the way.

Reply:
You can check the tests I added, I am not sure if I understand your question? You mean if the method counts the samples as failed when they have a status like library prep that is assigned after the qc was succesfull?

Reply:
Good point, thanks for pointing it out, I adjusted the implementation